test/blocklib/core/selector.hpp

```diff
@@ -0,0 +1,158 @@
+                // make the implicit consume all available behaviour explicit
+                std::for_each(ins.begin(), ins.end(), [](auto *input) { std::ignore = input->consume(input->available()); });
+            }
+            return work_return_status_t::OK;
+        }
+
+        std::set<std::size_t> used_inputs;
+
+        if (const auto select_available = select->available(); select_available > 0) {
+            auto select_span = select->get(select_available);
+            _selectedSrc     = select_span.back();
+            std::ignore      = select->consume(select_available); // consume all samples on the 'select' streaming input port
+        }
+
+        for (const auto &[input_index, output_indices] : _internalMapping) {
+            auto *input_reader    = ins[input_index];
+            auto  input_available = input_reader->available();
+
+            for (const auto output_index : output_indices) {
+                auto *writer = outs[output_index];
+                if (input_available > writer->available()) {
+                    input_available = writer->available();
+                }
+            }
+
+            if (_selectedSrc == input_index) {
+                if (input_available > monOut->available()) {
+                    input_available = monOut->available();
+                }
+            }
+
+            if (input_available == 0) {
+                continue;
+            }
+
+            auto copy_to_output = [&](auto &input_span, auto *output_writer, auto output_index) {
+                auto output_span = output_writer->reserve_output_range(input_available);
+
+                for (std::size_t i = 0; i < input_span.size(); ++i) {
+                    output_span[i] = input_span[i];
+                }
+
+                output_span.publish(input_available);
+            };
+
+            auto input_span = input_reader->get(input_available);
+            for (const auto output_index : output_indices) {
+                auto *output_writer = outs[output_index];
+                copy_to_output(input_span, output_writer, output_index);
+            }
+
+            if (_selectedSrc == input_index) {
+                copy_to_output(input_span, monOut, -1);
+            }
+
+            std::ignore = input_reader->consume(input_available);
+            used_inputs.insert(input_index);
+        }
+
+        for (auto src_port = 0U; src_port < ins.size(); ++src_port) {
+            if (used_inputs.contains(src_port)) continue;
+
+            if (backPressure) {
+                std::ignore = ins[src_port]->consume(0_UZ);
+
+            } else {
+                // make the implicit consume all available behaviour explicit
+                std::ignore = ins[src_port]->consume(ins[src_port]->available());
+            }
+        }
+
+        // N.B. some corner case-handling to be added:
+        // * one input mapped to multiple outputs -> check for consistent and produce the same min available() for all mapped outputs
```

Comment:
The `monitorOut` logic seems to have disappeared.
The described corner case doesn't seem to be handled (yet).

Reply:
monitor out is unified with the rest.

One input to multiple outputs works - there is a test for that, and for many inputs to one output.